src/Actions/Product/UpdateProduct.php

```diff
@@ -85,6 +90,30 @@ public function performAction(): Model
                 );
         }
 
+        if ($productCrossSellings !== null) {
+            $activeProductCrossSellings = [];
+            foreach ($productCrossSellings as $productCrossSelling) {
+                $productCrossSelling['product_id'] = $product->id;
+                if ($productCrossSelling['id'] ?? false) {
+                    $action = UpdateProductCrossSelling::make($productCrossSelling);
+                } else {
+                    $action = CreateProductCrossSelling::make($productCrossSelling);
+                }
+
+                $activeProductCrossSellings[] = $action->checkPermission()->validate()->execute()->id;
+            }
+
+            $removedProductCrossSellings = $product->productCrossSellings
+                ?->whereNotIn('id', $activeProductCrossSellings)
+                ->pluck('id')
+                ->toArray();
+            foreach ($removedProductCrossSellings as $removedProductCrossSelling) {
+                DeleteProductCrossSelling::make(['id' => $removedProductCrossSelling])
+                    ->checkPermission()
```

Comment:
really need to check the permission for every deletion? Also if an exception is thrown this can cause some confusion for the user, because the product is already updated at this point.